workspaces/manage/plugins/manage-module-tech-insights/src/columns/hooks.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useMemo } from 'react';
+
+import { useApi } from '@backstage/core-plugin-api';
+import { Entity } from '@backstage/catalog-model';
+import { techInsightsApiRef } from '@backstage-community/plugin-tech-insights';
+import type { Check } from '@backstage-community/plugin-tech-insights-common/client';
+
+import { useManageTechInsights } from '../components/ManageProvider';
+import { filterEmptyChecks, stringifyCheck } from '../utils';
+
+export function useEntityInsights(
+  entities: Entity[],
+  checkFilter: ((check: Check) => boolean) | undefined,
+  showEmpty: boolean,
+) {
+  const { allChecks, bulkCheckResponse, renderers } =
+    useManageTechInsights(checkFilter);
+  const techInsightsApi = useApi(techInsightsApiRef);
+
+  const { usedChecks, responses } = useMemo(
+    () => filterEmptyChecks(bulkCheckResponse, entities, showEmpty),
+    [bulkCheckResponse, entities, showEmpty],
+  );
+
+  const filteredChecks = allChecks.filter(check =>
+    showEmpty ? true : usedChecks.has(stringifyCheck(check)),
+  );
+
+  const uniqness = filteredChecks
+    .map(check => stringifyCheck(check))
+    .join(' $$ ');
+
+  const memoedFilteredChecks = useMemo(
+    () => filteredChecks,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [uniqness],
+  );
```

Comment:
probably removing all these memoization won't add any overhead but at least the code will be simpler?

Reply:
True, only a small improvement if the table cells would need to re-render, but the likelyhood of that is very small.